.github/scripts/add_footer.py

```diff
@@ -0,0 +1,92 @@
+from os import walk
+from os.path import join, relpath
+from utils import get_template
+from re import sub, search
+from urllib.parse import quote
+
+from utils import get_root_of_vault
+
+# These are directories and files to exclude
+# By adding a dir/file, this script will ignore them and never change them!
+DIRECTORIES_TO_EXCLUDE = ['.git', '.github', '.idea', 'venv', 'DO NOT COMMIT']
+FILES_TO_EXCLUDE = ['.DS_Store', '.gitignore']
+
+
+def add_footer(root: str, debug: bool = True):
+    """
+    Walks through the filetree rooted at `root`.
+    For each markdown file that it finds, it replaces a particular comment line with the corresponding template.
+
+    Parameters:
+        root: path from which this method should run. Generally: root of the hub.
+        debug: boolean that indicates wether or not to print logging statements
+    """
+    # Grab the template
+    template = get_template("footer")
+
+    # This is the regex to search for
+    # Note that we select the comment itself, and then ANYTHING afterwards
+    # This requires the "DOTALL" (?s) and "MULTILINE" (?m) flags to be set
+    comment = r"(?sm)%% Hub footer: Please don't edit anything below this line %%.*"
+
+    # Loop through the files
+    for root, dirs, files in walk(root, topdown=True):
+        # Exclude directories and files
+        dirs[:] = [d for d in dirs if d not in DIRECTORIES_TO_EXCLUDE]
+        files[:] = [f for f in files if f not in FILES_TO_EXCLUDE]
+
+        # Loop through the files
+        for file in files:
+            # We only care about markdown files
+            # Note: Alternative implementation is to use os.splitext;
+            # both work for this usecase
+            if file.endswith(".md"):
+
+                # Get the ABSOLUTE filepath
+                the_file = join(root, file)
+
+                if debug:
+                    print(f"Processing '{relpath(the_file, root)}'...")
```

Comment:
For future safety, I think it would be helpful to create a variable with this result of `relpath(the_file, root)` - that is then used in the two places.

This will guard against the possibility of one of the relpath calls being changed in future and not the other, after which the logging would then not represent that actual behaviour....